ppmclibs/tinycv_impl.cc

```diff
@@ -187,10 +187,13 @@ std::vector<int> search_TEMPLATE(const Image* scene, const Image* object,
     }
 
     // Optimization -- Search close to the original area working with ROI
-    int scene_x = std::max(0, int(x - margin));
-    int scene_y = std::max(0, int(y - margin));
-    int scene_bottom_x = std::min(scene->img.cols, int(x + width + margin));
-    int scene_bottom_y = std::min(scene->img.rows, int(y + height + margin));
+    // Scale the possition if object and scene have different sizes
+    int scaled_x = x * scene->img.cols / object->img.cols;
+    int scaled_y = y * scene->img.rows / object->img.rows;
```

Comment:
In theory it would make sense to avoid the implicit narrowing conversion here:
```suggestion
    long scaled_x = x * scene->img.cols / object->img.cols;
    long scaled_y = y * scene->img.rows / object->img.rows;
```
(In practice it likely doesn't matter because we'll run into an implicit narrowing conversion later anyway and there are also many explicit narrowing conversions.)